source/uwp/SharedRenderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -599,7 +603,11 @@ namespace AdaptiveCards::Rendering::Xaml_Rendering
     winrt::fire_and_forget XamlBuilder::SetSvgUriSource(winrt::SvgImageSource const imageSource,
                                                         winrt::Uri const uri)
     {
+#ifdef USE_WINUI3
+        co_await wil::resume_foreground(imageSource.DispatcherQueue());
+#else
         co_await winrt::resume_foreground(imageSource.Dispatcher());
+#endif
```

Comment:
Can these lines of code be consolidated into a routine? That way, if we need to make changes, they can be made in one location.

Reply:
@jwoo-msft Coroutines are not my strong suit... do you know what this would look like? I played around a bit but am not sure the correct way to do this with the co_awaits.

Reply:
I think we still need to detect running winui3, since we still need to use Dispatcher vs DispatcherQueue (DQ is not available on uwp [link](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.imaging.svgimagesource?view=winrt-22621)).

As for using WIL for both, UWP really doesn't seem to want to find that wil/cppwinrt_helpers.h file. I included the WIL nuget package, but still nothing. If we have to do the winui3 check anyway, it may not be worth the trouble.

Reply:
@krschau, I'm pretty sure you've already tried. what kind of issue did you ran when the wrapper routine returns a co-await able object, and co-await on the returned object?

Reply:
We can also create a macro or an inline function that takes no arguments. As we all agree, it is important to keep it in one place to avoid unnecessary time spent on maintenance. In this situation, using a macro seems appropriate. It is simply a single line of code, and it does not affect debuggability since it involves an API call to an external library. ACResumeForeGround();

Reply:
Originally I was getting something like [this](https://devblogs.microsoft.com/oldnewthing/20210809-00/?p=105539) but the suggested solution wasn't working for me. I think I'm getting farther this time...

Reply:
@jwoo-msft Very weird -- I tried adding this method and co_await-ing it in the rest of the code. But with this change, I get a `System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'` when rendering a card with an SVG.

```
inline auto ACResumeForeground(winrt::SvgImageSource const imageSource)
{
#ifdef USE_WINUI3
    return wil::resume_foreground(imageSource.DispatcherQueue());
#else
    return winrt::resume_foreground(imageSource.Dispatcher());
#endif
}
```

Reply:
Today is Windows Learning day :-) I think the quick solution here is using the macro. we just let the preprocess expand it. 
for the inline, what if the compiler did inline expansion, and wonder what it would look like after the expansion.

Reply:
@krschau 
you can also try something like this.
```
    auto inline GetDispatcher(winrt::SvgImageSource const &imageSource)
    {
	#ifdef USE_WINUI3
		return imageSource.DispatcherQueue();
	#else
		return imageSource.Dispatcher();
	#endif
    }
```
and call the inline where resume_foreground is called
```
co_await winrt::resume_foreground(GetDispatcher(imageSource));
```
this works.